internal/service/thirdpartyintegration/data_source_third_party_integrations.go

```diff
@@ -65,30 +65,6 @@ func flattenIntegrations(d *schema.ResourceData, integrations *matlas.ThirdParty
 
 func integrationToSchema(d *schema.ResourceData, integration *matlas.ThirdPartyIntegration) map[string]any {
 	integrationSchema := schemaToIntegration(d)
-	if integrationSchema.APIKey == "" {
-		integrationSchema.APIKey = integration.APIKey
-	}
-	if integrationSchema.ServiceKey == "" {
-		integrationSchema.APIKey = integration.ServiceKey
-	}
-	if integrationSchema.APIToken == "" {
-		integrationSchema.APIKey = integration.APIToken
-	}
-	if integrationSchema.RoutingKey == "" {
-		integrationSchema.APIKey = integration.RoutingKey
-	}
-	if integrationSchema.Secret == "" {
-		integrationSchema.APIKey = integration.Secret
-	}
-	if integrationSchema.Password == "" {
-		integrationSchema.APIKey = integration.Password
-	}
-	if integrationSchema.UserName == "" {
-		integrationSchema.APIKey = integration.UserName
```

Comment:
this implementation is clearly not in a good state (see that `APIKey` is set with unrelated properties), after adding testing verified this logic is not needed.